### PR TITLE
epoch key only proof

### LINKED ISSUE #159: Epoch key proof without state tree
There are instances where we want to prove control of an epoch key, without proving membership in a state tree. For example, proving control of a historical epoch key may be done without a state tree membership proof.

We only care about state tree membership in the case of non-repudiability. If a user proves an epoch key in the current epoch, but doesn't prove state tree membership, then they don't _have_ to receive reputation sent to them.

### PR DESCRIPTION
**Pull request by vimwitch**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Closes #159 

Depends #216 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/circuits/test/epochKeyLite.test.ts
Comment (vivianjeng) [inline packages/circuits/test/epochKeyLite.test.ts:89]: can the shift bits be more readable?
e.g.
```ts
(BigInt(attesterId) << BigInt( EPOCH_BITS + NONCE_BITS )) +
BigInt(epoch) << BigInt(NONCE_BITS))
```
or it can be concatenated
```ts
const data = revealNonce +
             attesterId.padStart(ATTESTER_ID_BITS) + 
             epoch.padStart(EPOCH_BITS) + ...
```

**Review by vivianjeng — CHANGES_REQUESTED** [review]
(haven't finished reviewed yet, but some quick comments)

File: packages/circuits/test/epochKeyLite.test.ts
Comment (vimwitch) [reply packages/circuits/test/epochKeyLite.test.ts:89]: Yea using constants defined in the proof class would make this more clear.

Comment (vimwitch) [reply packages/circuits/test/epochKeyLite.test.ts:89]: `padStart` shouldn't work, it has to be a numerical addition

Comment (vimwitch) [reply packages/circuits/test/epochKeyLite.test.ts:89]: This is out of scope for this pull request, documented in #238

File: packages/contracts/test/verifier.test.ts
Comment (vivianjeng) [inline packages/contracts/test/verifier.test.ts:65]: can use snapshot to revert evm like #243

Comment (vimwitch) [reply packages/contracts/test/verifier.test.ts:65]: Ah the verifier tests are mostly stateless. I'm not opposed to adding snapshot/revert logic, but i'd rather not block this pr and #239. Need these for `2.0.0-alpha-3` and zkhack next week.

Comment (vivianjeng) [reply packages/contracts/test/verifier.test.ts:65]: I am also good to go without snapshot logic
or after this is merged I can change it in #239

**Review by vivianjeng — APPROVED** [review]

File: packages/contracts/test/verifier.test.ts
Comment (vimwitch) [reply packages/contracts/test/verifier.test.ts:65]: Ok, there shouldn't be any conflicts. It should be possible to copy/paste the hooks into this `describe` block.

**Pull request merged**
